### PR TITLE
Icap whitelist

### PR DESCRIPTION
**Pull request by kekek2**
Exclude domains from antivirus check.

### CODE REVIEW

File: src/opnsense/mvc/app/controllers/OPNsense/Proxy/forms/main.xml [hunk 1]
```diff
@@ -537,6 +537,24 @@
                 <allownew>true</allownew>
                 <advanced>true</advanced>
             </field>
+            <field>
+                <id>proxy.forward.icap.whiteList</id>
+                <label>Whitelist</label>
```
Comment (fabianfrz) [inline src/opnsense/mvc/app/controllers/OPNsense/Proxy/forms/main.xml]: I would rename that as "Whitelist" can be misleading. Basically ICAP is some kind of an remote procedure call protocol and not a ACL enforcement tool (even if it's most often used for that). I would prefer something like "Exclusion List" which fits the use case better (traffic will not be sent to the ICAP server).

**Comment by mimugmail** [comment]
I applied the patches and did some testing, works good 👍

**Comment by fichtner** [comment]
This may be relevant as well while we're at it:

https://forum.opnsense.org/index.php?topic=5866.msg24763#msg24763

**Comment by fabianfrz** [comment]
@fichtner I took the link from the config for the docs:
https://github.com/opnsense/plugins/blob/master/www/c-icap/src/opnsense/service/templates/OPNsense/CICAP/virus_scan.conf#L4

**Comment by fabianfrz** [comment]
ups - my mistake, we should change the default settings in squid

**Comment by fabianfrz** [comment]
see #1822 - please note that this will be only used for new installations or installations, where the config of the proxy was never saved.

**Comment by fichtner** [comment]
@fabianfrz that's alright, we worry about the future, not the past 👍

**Comment by fichtner** [comment]
also added to 17.7.4, thanks!

**Pull request merged**